### PR TITLE
Remove `send_to_socket::Actor`

### PR DESCRIPTION
**Pull request by thomaseizinger**
This removes a level of indirection for actually sending messages which
simplifies the overall handling of the connection and unifies how
heartbeats are sent to the taker. This is important because we don't
want heartbeats to be handled in a special way as we use them to detect
issues with the connection.

May help with https://github.com/itchysats/itchysats/issues/759.

### CODE REVIEW

File: daemon/src/maker_inc_connections.rs [hunk 1]
```diff
@@ -107,7 +105,26 @@ pub struct Actor {
...
+/// A connection to a taker.
+struct Connection {
+    taker: Identity,
+    write: wire::Write<wire::TakerToMaker, wire::MakerToTaker>,
+    _tasks: Tasks,
+}
+
+impl Connection {
+    async fn send(&mut self, msg: wire::MakerToTaker) -> Result<()> {
+        let msg_str = msg.to_string();
```
Comment (bonomat) [inline daemon/src/maker_inc_connections.rs:119]: Shall we add a trace here to log what message is being send?

Comment (thomaseizinger) [reply daemon/src/maker_inc_connections.rs:119]: Addressed in https://github.com/itchysats/itchysats/pull/927/commits/21b20f6fdbff870393cd2f92a9d2caee9bfda50c.

File: daemon/src/maker_inc_connections.rs [hunk 2]
```diff
@@ -118,6 +118,8 @@ impl Connection {
     async fn send(&mut self, msg: wire::MakerToTaker) -> Result<()> {
         let msg_str = msg.to_string();
 
+        tracing::trace!(target = "wire", taker_id = %self.taker, "Sending {}", msg_str);
```
Comment (bonomat) [inline daemon/src/maker_inc_connections.rs:121]: nice, that's useful 👍

**Review by bonomat — APPROVED** [review]
Awesome, thanks.

**Comment by thomaseizinger** [comment]
Let's see how it performs on testnet.

**Comment by thomaseizinger** [comment]
Just stress testing this against the testnet maker and works like charm so far.

**Pull request merged**